### ISSUE TITLE
Item Trade drop restriction cannot be overwriten by GM level

### ISSUE THREAD

**Issue by mazvi** [rathena/rathena#2533]
* **rAthena Hash**: caf3152

* **Client Date**: 20170517RE

* **Server Mode**: RENEWAL

* **Description of Issue**: Item Trade GM Group Level Overide Not Work
  * Result:  iam GM 99, on item trade i set grup 90 but the item canot be droped by GM 99
  * Expected Result: 
  * How to Reproduce: 
  * Official Information:
  

* **Modifications that may affect results**:

**Comment by Jeybla** [comment]
Could be an issue with the import table. Did you add it to the import table or did you edit the actual line?

**Comment by mazvi** [comment]
true iam using import folder like this /rathena/db/import/item_trade.txt

for databse item iam using txt not sql

below my code

`// Item Trade Restrictions Database
// Defines special trade rules for individual items.
//
// Structure of Database:
// Item ID, TradeMask,Group Level Override
//
// Legend for 'TradeMask' field (bitmask):
// 1   - item can't be dropped
// 2   - item can't be traded (nor vended)
// 4   - wedded partner can override restriction 2
// 8   - item can't be sold to npcs
// 16  - item can't be placed in the cart
// 32  - item can't be placed in the storage
// 64  - item can't be placed in the guild storage
// 128 - item can't be attached to mail
// 256 - item can't be auctioned
//
// Example:
// 1161,67,50
//	Balmung cannot be dropped, traded, or placed in Guild Storage (1+2+64).
//	Only groups of group-level 50 and up can override this setting.

//475 Hanya Bisa Storage
//507 Tidak Bisa Semua
//499 Bisa dijual di NPC

64307,475,100 //ExtraJozz
64340,507,90 //Gashat

`

the item 64340 canot droped by GM Grup Level 99

**Comment by Jeybla** [comment]
Should be fixed together with #2503

Should we also have the item_trade db in YAML format?

**Comment by mazvi** [comment]
I dont understand what you mean Should we also have the item_trade db in YAML format?

**Comment by Lemongrass3110** [comment]
This is not related to the import at all.
This happens due to an additional check of drop permissions in map_addflooritem.

```
if (!(flags&4) && battle_config.item_onfloor && (itemdb_traderight(item->nameid)&1))
		return 0; //can't be dropped
```

At this point we do not have the player session accessible and cannot compare the override GM level.

From my point of view this looks like a sanity check, if somebody forgot to add a pc_candrop call before calling this function. But there are a lot of usages of this function so we would have to check a lot of references to it.

**Comment by AzarthMZintos** [comment]
any news on this?

**Comment by gidzdlcrz** [comment]
I can confirm this still persist in the latest git.

**Comment by reunite-ro** [comment]
Anyone can share if they you guys already solved this issue ? Thanks.

**Comment by rlsbautista** [comment]
> Anyone can share if they you guys already solved this issue ? Thanks.

As far as I know this has already been reproduced. Do you still experience the issue?

**Comment by reunite-ro** [comment]
> > Anyone can share if they you guys already solved this issue ? Thanks.
> 
> As far as I know this has already been reproduced. Do you still experience the issue?

I saw this issues after reading some post on rAthena forum. Ive tried and its still not working if we use trade override gm level.

You can try it and see is it working or not.

**Comment by aleos89** [comment]
Fixed in 047e5ee.
